tsweb/jsonhandler_test.go

```diff
@@ -90,11 +90,11 @@ func TestNewJSONHandler(t *testing.T) {
 	h31 := JSONHandlerFunc(func(r *http.Request) (int, interface{}, error) {
 		body := new(Data)
 		if err := json.NewDecoder(r.Body).Decode(body); err != nil {
-			return http.StatusBadRequest, nil, err
+			return 0, nil, Error(http.StatusBadRequest, err.Error(), err)
```

Comment:
Replacing some of these tests is fine, but I'd like to see a couple of them duplicated with the old functionality to make sure they work.

Also add a new test that covers what happens if there are two status codes, one via HTTPError and the other by the return int. If they disagree (i.e. both a non-zero and non-equal) we should log an [unexpected].